src/Sylius/Bundle/CoreBundle/EventListener/ProductDeleteListener.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\EventListener;
+
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Sylius\Component\Core\Model\ProductInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class ProductDeleteListener
+{
+    /**
+     * @var ContainerInterface
+     */
+    protected $container;
+
+    /**
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
```

Comment:
Perhaps dependency resolution cycle! Because you cannot inject managers into doctrine listeners!